odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/mapper/odoo/SaleOrderLineMapper.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.mapper.odoo;
+
+import com.ozonehis.eip.odoo.openmrs.mapper.ToOdooMapping;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrderLine;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.Dosage;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class SaleOrderLineMapper<R extends Resource> implements ToOdooMapping<R, SaleOrderLine> {
```

Comment:
Is this class meant to be abstract? Why does it have a type argument?

Reply:
Because `R` can be an instance of `MedicationRequest` as well as `ServiceRequest`